internal/clients/kafka/acl/acl.go

```diff
@@ -162,8 +162,16 @@ func Generate(name string, params *v1alpha1.AccessControlListParameters) *Access
 }
 
 // LateInitializeSpec fills empty ACL spec fields with the data retrieved from Kafka.
-func LateInitializeSpec() bool {
-	lateInitialized := true
+func LateInitializeSpec(in *v1alpha1.AccessControlListParameters, observed *AccessControlList) bool {
+	lateInitialized := false
```

Comment:
We late initialize optional parameters to reflect their defaults on the server side. 

With ACL resources spec, we don't have any optional parameters, so, I don't think we need to late-init anything. Hence, we can simply return `lateInialized=false` in ExternalObservation.

Reply:
Hasan, this was actually my suggestion and misunderstanding.

We have a situation where this workflow is broken:  
User applies a yaml -> ACL gets created -> External name get set (to the applied yaml) -> User attempts to update (by applying new YAML) -> External name != new yaml spec in k8s -> K8s thinks its an update/the manifest mismatches

My idea was just to use the LateInitialize to set the k8s configuration back to the actual state of the Kafka server.

What would be the correct way to do this?

Reply:
I mean handling this mismatch in Observe and returning an error:

```
-> External name != new yaml spec in k8s -> Observe returns a proper error
```

Reply:
Yes, assuming you mean the annotation.

Reply:
Oh, I see.  

Yes, we're handling that mismatch and returning the error, but we weren't sure how to get it to not update.  Also, I believe the YAML  in k8s needs to be the returned to the original for the delete, otherwise, it won't delete.  So we have to set it "back" to the original YAML somehow.  

I'll use the annotation.  Thanks :)